Security Weapons Vendor

mybluecorners opened the pull request:
<!-- The text between the arrows are comments - they will not be visible on your PR. -->
<!-- To automatically tag this PR, add the uppercase label(s) surrounded by brackets below, for example: [LABEL] -->

## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->
Adds a vendor with a few selectable security loadouts.  loadouts currently include:

loadouts as of latest patch are:

taser+baton
wavegun+baton
baton, 2 robust donuts (synap,epi,sugar), 4 morphine auto injectors
taser shotgun, pepper grenades, and baton

'robust donut' is a new kind of donut that has some partial stun reducing + stam regenerating chems inside.

The intent is to replace the baton/taser weapon spawns in sec lockers with these vendors. The map changes and token spawns are not in this PR - I will make those updates when this is merged.


## Why's this needed? <!-- Describe why you think this should be added to the game. -->
Sec gear, primarily ranged stun weapons + batons, are very strong weapons. They are high priority items for an antagonist (especially one who is trying to murder rampage) to pick up near the start of the round. It's usually quite easy to get multiple tasers and batons (empty the sec lockers and stuff your bag!) considering there are more than enough to go around when security isn't fully staffed.
I think it's better if these weapons are in play only when a security officer invokes them. In order to acquire this gear as an antagonist you'll have to kill or steal from security first.

Additonally, security gets some new options for weapons they can use at round start that were previously unavailable due to balance concerns from them having so many weapons already. Instead of picking up a taser, they can now choose guns like a wavegun or phaser or taser shotgun.

Carbadox commented:
what's gonna happen with the security lockers or the sec loadout crate from QM?

mybluecorners commented:
security lockers would have a couple weapons removed but stay the same otherwise. The sec loadout crate would go unchanged (QMs can already get a lot of the equipment in these loadouts already)

jaffjafferson commented:
maybe swap the phaser in offense for a laser, or a clock 17? a single phaser is incredibly weak when compared to the other 3

Kyle2143 commented:
We were talking about this in admin char eariler but I'll put it here for posterity.

We kinda sorta agreed that a good change in conjunction with this is to bump up the tasers' standard power cell from 100pu to 200pu and increase the cost of taser bolts so that it sould give you 8-9 shots for a 200pu cell (compared to 6 shots for a 100pu cell currently).

And bump up the security pouch's 200pu cell to be 300pu.

And change phasers to work a bit more like how they used to work in the distant past so you can set a phaser's lethality from 0 to 100%  where it will do stuns and some damage when you have it set to less than 100% damage.

Kyle2143 commented:
This is an unrelated suggestion, might not actually be good, I'm uncertain.

Allow the vender to take the items for a loadout back from a player and let them choose new items if the player presents it with all the items for a class.

TDHooligan commented:
I like the idea, though i'll list some things that come to mind to _possibly_ be looked into in case rampagers find their way around sec being empty.

Traitors, lings and werewolves have access to powerful tools that sec's equipment is there to deal with in the first place. Maybe look into:
- more readily available consumable stun equipment (like plates, stungloves), Crew usually need at most 1-2 stuns, and are generally happy to carry a banana peel/plate, while rampage traitors will need to clog inventory space to carry a huge number.
- toning down/restricting the early-game power of rampage-oriented traitor weapons, like a time lock for SPAS-12, horror arm, wolves

There is also still an easy source for stun batons - Security bots, which are beneficial for traitors to break in general, and a net loss for the crew should they smash it. Traitors may target secbots more, so maybe their drops should be adjusted.

Perhaps a revision of how many sunglasses/engi helms are around may also be in order? Flashes are still functionally stun batons and will most likely take over the role of the stun baton very quickly once people catch on to them.

Common roles that can play as backup security roles owing to their access (Captain, HoP) could also require some some sort of equipment buff (though Captain does have an E-gun, perhaps HoP should get a taser)

Carbadox commented:
So a couple of things that bother me.

The general premise seems pretty limiting. Having kits be limited to 1 belt, 1 baton and X item seems pretty awkward to me. I think the buy list should be consolidated into accesories and main items. Accessories would be free and be things like belts, SecHuds, etc. Then have the main items cost points. Stun batons, tasers, phasers, etc. I think this allows people to still have a measure in freedom of picking their loadout. 2 lethal guns without a baton might not be the best strategy but someone might want to do.

A phaser seem pretty weak and bit unfitting for Security, I think replacing it with the Clock or something similar would be a better alternative. 
Removing SecBots dropping stun batons on death seems in line with the general idea of this PR
Not entirely relevant but rigged light tubes have a (100%?) high chance of breaking open crates. I think is unintended but if it is, it's quite broken since it makes breaking locked crates open trivial. I'm fine with crates being easy to break open as long as it's done in a proper fashion EG crates having a health bar and dealing any sort of damage to them will crack them open.

PrimeNumb commented:
How are tokens meant to be distributed? Are security officers gonna spawn with one in their inventory?

Tarmunora commented:
I disagree with the idea of giving security a lethal-option loadout as part of base kit, and strongly disagree with giving them a kinetic weapon. It feels highly off-brand and generally gives me bad vibes

PrimeNumb commented:
> I disagree with the idea of giving security a lethal-option loadout as part of base kit, and strongly disagree with giving them a kinetic weapon. It feels highly off-brand and generally gives me bad vibes

Yeah, this. The offensive loadout mentioned here would be largely unusable, since you're not meant to lethal crimers unless absolutely necessary, and it might give the wrong impression for newers players that think it's okay to do that.

Tarmunora commented on code/modules/food_and_drink/snacks.dm:
```diff
@@ -1120,6 +1120,13 @@
+		icon_state = "donut4"
+		initial_volume = 30
+		initial_reagents = list("sugar"=10,"synaptizine"=10,"epinephrine"=10)
+
```
Might be worth boosting the bite count and/or heal_amt on these

Tarmunora commented:
Can't say I'm a fan of the 'sidearm' stuff in here. EOD is pretty specialized (and only gives the suit and no helmet?) and doesn't really make sense as a sidearm, flaregun see above comment wrt lethals. Barrier is the only one there that makes sense to me, and could just as well be left in the lockers.

Gannets commented:
I think this is a neat idea and I'm glad my vendors could be used elsewhere. The concept of sec classes is a good one though maybe there's room to tweak exactly what comes in each kit.

TDHooligan commented:
if we remove stunbatons from some security officers it's going to become important that sec retain the ability to hold a taser victim down, perhaps cuffing with sec training should automatically pin downed victims? means there will be fewer stunbatons kicking about

Ley-bunned commented:
Fog grenades are pretty bad for sec because the only folk that can see through them are traitor thermals. I suggest crowd control grenades instead.

mybluecorners commented:
I changed all loadouts to include a baton + a main weapon. Wavegun moved to offense, removed phaser. Support doesn't get a weapon aside from baton, instead they get a few morphine autoinjectors along with their robust donuts. 

Also removed 'sidearm' stuff, barrier would remain in the sec lockers.
